### PR TITLE
Add redirect for moved and renamed files

### PR DESCRIPTION
**Pull request by Fryuni**
Redirects for all the changes made during the Talking and Doc'ing today on #6620

### CODE REVIEW

File: vercel.json [hunk 1]
```diff
@@ -5,6 +5,48 @@
...
+    },
+    {
+      "source": "/:lang/core-concepts/framework-components",
+      "destination": "/:lang/guides/framework-components"
+    },
+    {
+      "source": "/:lang/core-concepts/sharing-state",
+      "destination": "/:lang/recipes/sharing-state-islands"
+    },
+
```
Comment (TheOtterlord) [inline vercel.json:49]: The redirects don't seem to be working in the preview. I have a feeling we just need the trailing slash from the full path:
```suggestion
    {
      "source": "/:lang/core-concepts/project-structure/",
      "destination": "/:lang/basics/project-structure/"
    },
    {
      "source": "/:lang/core-concepts/astro-components/",
      "destination": "/:lang/basics/astro-components/"
    },
    {
      "source": "/:lang/core-concepts/astro-pages/",
      "destination": "/:lang/basics/astro-pages/"
    },
    {
      "source": "/:lang/core-concepts/layouts/",
      "destination": "/:lang/basics/layouts/"
    },
    {
      "source": "/:lang/core-concepts/astro-syntax/",
      "destination": "/:lang/basics/astro-syntax/"
    },
    {
      "source": "/:lang/core-concepts/rendering-modes/",
      "destination": "/:lang/basics/rendering-modes/"
    },

    {
      "source": "/:lang/core-concepts/routing/",
      "destination": "/:lang/guides/routing/"
    },
    {
      "source": "/:lang/core-concepts/endpoints/",
      "destination": "/:lang/guides/endpoints/"
    },
    {
      "source": "/:lang/core-concepts/framework-components/",
      "destination": "/:lang/guides/framework-components/"
    },
    {
      "source": "/:lang/core-concepts/sharing-state/",
      "destination": "/:lang/recipes/sharing-state-islands/"
    },
```

**Review by TheOtterlord — COMMENTED** [review]
Thanks @Fryuni!

File: vercel.json (hunk 1 above)
Comment (sarah11918) [reply vercel.json:49]: I'll also ask @delucis to check on these!

**Review by TheOtterlord — APPROVED** [review]
Confirmed all redirects work as expected :partying_face:

**Pull request merged**